clang/lib/StaticAnalyzer/Checkers/StreamChecker.cpp

```diff
@@ -696,6 +730,69 @@ struct StreamOperationEvaluator {
 
 } // end anonymous namespace
 
+//===----------------------------------------------------------------------===//
+// Definition of NoStreamStateChangeVisitor.
+//===----------------------------------------------------------------------===//
+
+namespace {
+class NoStreamStateChangeVisitor final : public NoOwnershipChangeVisitor {
+protected:
+  /// Syntactically checks whether the callee is a freeing function. Since
+  /// we have no path-sensitive information on this call (we would need a
+  /// CallEvent instead of a CallExpr for that), its possible that a
+  /// freeing function was called indirectly through a function pointer,
+  /// but we are not able to tell, so this is a best effort analysis.
+  bool isFreeingCallAsWritten(const CallExpr &Call) const {
+    const auto *StreamChk = static_cast<const StreamChecker *>(&Checker);
+    if (StreamChk->FCloseDesc.matchesAsWritten(Call))
+      return true;
+
+    return false;
```

Comment:
```suggestion
    return StreamChk->FCloseDesc.matchesAsWritten(Call));
```

But given the complexity of the `FCloseDesc` member, I'd simply suggest inlining here. That would enable us to omit the static cast that makes this a bit ugly.

Reply:
The retionale behind not inlining this is that we could add more functions that could close the stream (annotations, maybe). The checker object is owned by the base class, so the static cast is necessary.